src/parser/internal.ts

```diff
@@ -105,8 +106,27 @@ function parseNode(node: md.FlowContent): notion.Block[] {
     case 'heading':
       return [parseHeading(node)];
 
-    case 'paragraph':
-      return [parseParagraph(node)];
+    case 'paragraph': {
+      const imageElement = node.children.filter((child) => child.type === 'image')
+      if (imageElement.length) {
+        node.children.forEach((child, index) => {
+          if (child.type === 'image') {
+            delete node.children[index];
+          }
+        })
+        const imageBlocks = imageElement.map((element) => {
+          if (element.type === 'image') {
+            return notion.image(element.url)
+          } else {
+            return notion.image('')
+          }
+        })
+        return [...imageBlocks, parseParagraph(node)]
```

Comment:
What do you think about pulling this out into a separate case and the logic into its own function for parsing images so we can maintain the simple style here?